src/main/java/org/tugraz/sysds/hops/BinaryOp.java

```diff
@@ -467,11 +470,11 @@ else if(et == ExecType.SPARK)
 							getDataType(), getValueType(), et);
 				}
 				else if (mbin == MMBinaryMethod.MR_BINARY_M) {
-					boolean partitioned = false;
```

Comment:
That's a good catch as this was only necessary for map reduce - so maybe remove the argument from the BinaryM lop to get rid of it once and for all.